lang/csharp/src/apache/test/AvroGen/AvroGenTests.cs

```diff
@@ -0,0 +1,664 @@
+  ]
+}";
+
+        private const string _stringLogicalType = @"
+{
+  ""namespace"": ""org.apache.avro.codegentest.testdata"",
+  ""type"": ""record"",
+  ""name"": ""StringLogicalType"",
+  ""doc"": ""Test logical type applied to field of type string"",
+  ""fields"": [
+    {
+      ""name"": ""someIdentifier"",
+      ""type"": {
+        ""type"": ""string"",
+        ""logicalType"": ""uuid""
+      }
+},
+    {
+    ""name"": ""someJavaString"",
+      ""type"": ""string"",
+      ""doc"": ""Just to ensure no one removed <stringType>String</stringType> because this is the basis of this test""
+    }
+  ]
+}";
+
+        private Assembly TestSchema(
+            string schema,
+            IEnumerable<string> typeNamesToCheck = null,
+            IEnumerable<KeyValuePair<string, string>> namespaceMapping = null,
+            IEnumerable<string> generatedFilesToCheck = null)
+        {
+            // Create temp folder
+            string outputDir = AvroGenHelper.CreateEmptyTemporyFolder(out string uniqueId);
+
+            try
+            {
+                // Save schema
+                string schemaFileName = Path.Combine(outputDir, $"{uniqueId}.avsc");
+                System.IO.File.WriteAllText(schemaFileName, schema);
+
+                // Generate from schema file
+                Assert.That(AvroGenTool.GenSchema(schemaFileName, outputDir, namespaceMapping ?? new Dictionary<string, string>()), Is.EqualTo(0));
+
+                // Check if all generated files exist
+                if (generatedFilesToCheck != null)
+                {
+                    foreach (string generatedFile in generatedFilesToCheck)
+                    {
+                        Assert.That(new FileInfo(Path.Combine(outputDir, generatedFile)), Does.Exist);
+                    }
+                }
+
+                // Compile into netstandard library and load assembly
+                Assembly assembly = AvroGenHelper.CompileCSharpFilesIntoLibrary(
+                    new DirectoryInfo(outputDir)
+                        .EnumerateFiles("*.cs", SearchOption.AllDirectories)
+                        .Select(fi => fi.FullName),
+                        uniqueId);
+
+                if (typeNamesToCheck != null)
+                {
+                    // Check if the compiled code has the same number of types defined as the check list
+                    Assert.That(typeNamesToCheck.Count(), Is.EqualTo(assembly.DefinedTypes.Count()));
+
+                    // Check if types available in compiled assembly
+                    foreach (string typeName in typeNamesToCheck)
+                    {
+                        Type type = assembly.GetType(typeName);
+                        Assert.That(type, Is.Not.Null);
+
+                        // Instantiate
+                        object obj = Activator.CreateInstance(type);
+                        Assert.That(obj, Is.Not.Null);
+                    }
+                }
```

Comment:
do we care about the above CodeQL suggestion ?

Reply:
What it tries to suggest is that the first 2limes of the foreach loop should be combined into foreach(Type type in typeNamesToCheck.Select(typeName => assembly.GetType(typeName)).

After that the loop enumerate the types which were created on the fly with the Select.

The reasons I don't like that suggestion is this. It is a unit test, and the test code readability and understand the test's intention is more important. Otherwise I have to decipher what the Select is creating.